### ISSUE TITLE
fastboot_orig oem unlock XXXXXXXX_sig.data does not advance and stays in downloading signature

### ISSUE THREAD

**Issue by jfredyp** [Maxpnl/Xiaomi-Unlocker-Fake-Fastboot#1]
I followed all the steps indicated but executing astboot_orig oem unlock XXXXXXXX_sig.data I have been stuck. Any ideas?

**Comment by Maxpnl** [comment]
It happened to me as well, just keep retrying until you are lucky, at least this is what worked for me, not sure if that's a general rule. 
In my case it worked after like trying this for 3-4 times (and one random reboot). If it hangs just kill the process (CTRL+C) and retry running the command, it will eventually work if you have my same problem. Let me know if you manage to do it :)

**Comment by jfredyp** [comment]
I will keep trying thanks for responding. An additional question arises: What should be the weight of XXXXXXXX_sig. ? data mine weighs 1MB

**Comment by Maxpnl** [comment]
Mine is 256B, but it could vary between devices, not sure, I have a POCO F1 btw. Oh and make sure that you get a new XXXX_sig.data everytime you reboot the phone because the token will be refreshed (it should be explained in the guide)

**Comment by Maxpnl** [comment]
1MB seems like a lot to me tbh, but I don't know how it's getting generated and what response each device is expectin, so can't tell if that's normal

**Comment by jfredyp** [comment]
thanks for your prompt response. Now I am getting this error trying to load the sig.data 

![image](https://user-images.githubusercontent.com/90223812/132267076-4690c838-7f97-4707-8eeb-4217992affb0.png)


Did it ever happen to you?

**Comment by Maxpnl** [comment]
Yes it happened to me as well, it worked for me after maybe 3-5 retries. Let me know if you manage to unlock it :)

P.S. out of curiosity, is your processor from the AMD Ryzen series? In case it is, which one do you have? I am trying to understand if the use cases where this tool actually helps are wider/narrower than the one I listed.

**Comment by Maxpnl** [comment]
my suggestion is just keep retrying, after a while you'll get lucky if you have the same problem I had

**Comment by jfredyp** [comment]
My processor is a ryzen threadripper 2920x. I've tried intel and it hasn't worked even on a mackbook, I'm losing my mind with this bootloader unlock

**Comment by sok904** [comment]
Maybe I´m wrong but it could be a chipset problem. I tried to solve they same issue as you and followed all the steps you provided, but the device wouldnt download the signature. I tried it about 20 Times but the only time I got a positive result, nothing has changed at all. The Solution was to plug in the Device in a USB-Port which is directly attached to the CPU rather then the Chipset. The Mi Unlock Tool worked instantly without any problem. I don`t have a way to proof my point, but maybe you want to take a look into it.

btw i have a Ryzen 5600X
